python/tk_desktop/desktop_engine_site_implementation.py

```diff
@@ -226,9 +232,24 @@ def run(self, splash, version, **kwargs):
         self._current_login = self._engine.sgtk.shotgun.find_one(
             "HumanUser",
             [["login", "is", human_user.login]],
-            ["id", "login"]
+            ["id", "login", "permission_rule_set"]
         )
 
+        # We've retrieved the permission rule set for immediate consumption, so
+        # copy the value and remove it from the user dictionary.
+        permission_rule_set = self._current_login[_PERMISSION_RULE_SET]
+        del self._current_login[_PERMISSION_RULE_SET]
+
+        # If the permission rule set is for the admin, we'll register a few
+        # commands.
+        if permission_rule_set["name"] == _ADMIN_RULE_SET:
+            if assign_users_to_template_project_command.is_available(self._engine):
```

Comment:
I am not super familiar with this code, but the class name says `DesktopEngineSiteImplementation`, implying that this is site specific. So do we need the `is_available()` check? Will it ever return false? If it does return false in some cases, a comment explaining could be helpful I think!